drivers/storage/portworx/portworx.go

```diff
@@ -408,6 +410,47 @@ func setNodeSpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
 	toUpdate.Spec.Nodes = updatedNodeSpecs
 }
 
+func setSecuritySpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
+	// all default values if one is not provided below.
+	defaultAuthTemplate := &corev1alpha1.AuthSpec{
+		GuestAccess: &corev1alpha1.GuestRoleEnabled,
+		Authenticators: &corev1alpha1.AuthenticatorsSpec{
+			SelfSigned: &corev1alpha1.SelfSignedSpec{
+				Issuer:        stringPtr("openstorage.io"),
+				TokenLifetime: metav1DurationPtr(time.Hour * 24),
+			},
+		},
+	}
+
+	if toUpdate.Spec.Security != nil {
+		if toUpdate.Spec.Security.Enabled == nil {
+			// not enabled by default
+			toUpdate.Spec.Security.Enabled = boolPtr(false)
+		} else if *toUpdate.Spec.Security.Enabled {
+			if toUpdate.Spec.Security.Auth != nil {
+				// security enabled and auth provided, add defaults if needed.
+				if toUpdate.Spec.Security.Auth.GuestAccess == nil {
+					toUpdate.Spec.Security.Auth.GuestAccess = defaultAuthTemplate.GuestAccess
+				}
+				if toUpdate.Spec.Security.Auth.Authenticators == nil || toUpdate.Spec.Security.Auth.Authenticators.SelfSigned == nil {
+					toUpdate.Spec.Security.Auth.Authenticators = defaultAuthTemplate.Authenticators
+				} else {
+					if toUpdate.Spec.Security.Auth.Authenticators.SelfSigned.Issuer == nil {
```

Comment:
Extending the comment from setJwtIssuer - 
If selfSigned.Issuer == nil, check env var and set if the value is present, else set the default.
Also let's remove the env right here so there are no duplicates in the future. That way we won't need the setJwtIssuer method.

As said before, in deployment.go where we form env vars for the pod, let's just add the issuer env from this value if security is enabled.

Reply:
Fixed this